python/tests/unit/arcticdb/version_store/test_empty_column_type.py

```diff
@@ -784,10 +789,31 @@ def test_date(self, lmdb_version_store_static_and_dynamic, date_dtype):
         lmdb_version_store_static_and_dynamic.update("sym", df)
         assert_frame_equal(lmdb_version_store_static_and_dynamic.read("sym").data, df)
 
-class TestEmptyTypesIsOverriden:
+
+class TestEmptyTypeIsOverriden:
+    """
+    When an empty column (or a column containing only None) values is initially written it is assigned the empty type.
+    The first write/update to change the type determines the actual type of the column. Test that the first non-empty
+    append determines the actual type and subsequent appends with different types fail.
+    """
+
+
     def test_cannot_append_different_type_after_first_not_none(self, lmdb_version_store_static_and_dynamic):
         lmdb_version_store_static_and_dynamic.write("sym", pd.DataFrame({"col": [None, None]}))
         lmdb_version_store_static_and_dynamic.append("sym", pd.DataFrame({"col": [1, 2, 3]}))
         lmdb_version_store_static_and_dynamic.append("sym", pd.DataFrame({"col": [None, None]}))
         with pytest.raises(Exception):
-            lmdb_version_store_static_and_dynamic.append("sym", pd.DataFrame({"col": ["some", "string"]}))
\ No newline at end of file
+            lmdb_version_store_static_and_dynamic.append("sym", pd.DataFrame({"col": ["some", "string"]}))
+
+    @pytest.mark.parametrize(
+            "incompatible_indexes",
+            [
+                (pd.RangeIndex(0,3), list(pd.date_range(start="1/1/2024", end="1/3/2024"))),
+                (list(pd.date_range(start="1/1/2024", end="1/3/2024")), pd.RangeIndex(0, 3))
+            ]
+    )
+    def test_cannot_append_different_index_type_after_first_non_empty(self, lmdb_version_store_static_and_dynamic, incompatible_indexes):
+        lmdb_version_store_static_and_dynamic.write("sym", pd.DataFrame({"col": []}))
+        lmdb_version_store_static_and_dynamic.append("sym", pd.DataFrame({"col": [1,2,3]}, index=incompatible_indexes[0]))
```

Comment:
We should read the data back here and have some assertions about the `dtype` of `col` and the type of the index